### PR TITLE
Added whitelisting feature to keeper

### LINKED ISSUE #206: Whitelisting collaterals
### Goal

Keeper user is able to specify collaterals for the keeper as env variable

### Context
Currently our [keeper bot](https://github.com/sidestream-tech/unified-auctions-ui/tree/main/bot#readme) checks and tries to participate in auctions for all supported collaterals. Same with the twitter bot. Instead, we want the possibility to specify which collaterals should be "whitelisted". This however should be an optional variable and the default behaviour should stay the same.

### Tasks
- [x] introduce new env variable for the bot, eg `ETH-A, MATIC-A`
- [x] extend readme with explanation and example
- [x] parse provided value and validate it (providing `ETH-X` or `FKSALJF` should throw an error on startup)
- [x] extend logs with whitelisting status (eg `auctions: fetching only ETH-A, MATIC-A auctions`)
- [x] test bot using [our hardhat setup](https://github.com/sidestream-tech/unified-auctions-ui/tree/main/hardhat#readme)

### PR DESCRIPTION
**Pull request by Tejas-Choudekar**
closes #206 

To do:
Write documentation in readme.md

### CODE REVIEW

**Comment by benediktkaiser** [comment]
Tested on Localhost. LGTM but further testing would be appreciated! 

Screenshot 2022-04-13 at 15 35 41

File: bot/src/keeper.ts [hunk 1]
```diff
@@ -40,6 +55,14 @@ const checkAndParticipateIfPossible = async function (auction: AuctionInitialInf
         return;
     }
 
+    // check if collateral is white listed
+    if (!isCollateralInWhitelist(auction.collateralSymbol)) {
+        console.info(
+            `keeper: auction "${auction.id}" was not executed because "${auction.collateralSymbol}" is not in the whitelist.`
+        );
+        return;
+    }
```
Comment (valiafetisov) [inline bot/src/keeper.ts:64]: As specified in the issue, this should be applied not only for the keeper, but for the twitter bot as well. Hint: we shouldn't even fetch auctions of the non-whitelisted collaterals

File: bot/src/keeper.ts [hunk 2]
```diff
@@ -7,10 +7,25 @@ import {
...
+} from './variables';
 
 let isSetupCompleted = false;
 const currentlyExecutedAuctions = new Set();
+function isCollateralInWhitelist(collateralSymbol: string): boolean {
+    if (!KEEPER_WHITELIST) {
+        return true;
+    }
+    // First parse the string and make it into an array
+    const whiteListArray = KEEPER_WHITELIST.split(',');
```
Comment (valiafetisov) [inline bot/src/keeper.ts:24]: Given the string `"ETH-A, MATIC-A"` it will produce `" MATIC-A"` which will not be equal to `"MATIC-A"`

**Review by valiafetisov — COMMENTED** [review]
Please check the tasks in the issue description that was addressed so far!

File: bot/src/keeper.ts (hunk 2 above)
Comment (valiafetisov) [reply bot/src/keeper.ts:24]: Also, the env variable is parsed on every run of the function and not on startup as requested

**Pull request closed without merging**